lib/tty/table/column_constraint.rb

```diff
@@ -141,7 +141,8 @@ def shrink
         ratio       = ((natural_width - renderer.width) / column_size.to_f).ceil
 
         widths = (0...column_size).reduce([]) do |lengths, col|
-          lengths + [renderer.column_widths[col] - ratio]
+          width = (renderer.column_widths[col] - ratio).clamp(minimum_width, renderer.width)
```

Comment:
Could you please replace `clamp` with ternary comparison? The reason being, for now I'd like to keep the gem working on older Rubies if possible and it seems `calmp` was added in `2.4.0`. Sorry for this restriction.

Reply:
Yeah! I worried that would be a problem. I'll replace it.

Reply:
I wish I could accept `clamp`- it's so much nicer and expressive.